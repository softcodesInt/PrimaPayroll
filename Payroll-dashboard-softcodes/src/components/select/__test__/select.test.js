/* -------------------------- External Dependencies ------------------------- */
import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

/* --------------------------- Internal Dependency -------------------------- */
import Select from '..';

const SelectMock = ({ ...rest }) => {
	return (
		<>
			<Select
				id="select"
				label="select"
				isInvalid={false}
				value="select"
				onChange={(e) => e.target.value}
				{...rest}
			>
				<option value="1">Option 1</option>
				<option value="2">Option 2</option>
				<option value="3">Option 3</option>
			</Select>
		</>
	);
};

describe('Select Component', () => {
	afterEach(cleanup);

	it('Select Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<SelectMock />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Select Renders Correctly With Attributes', () => {
		const { getByTestId } = render(<SelectMock />);
		// console.log();
		expect(getByTestId('select')).not.toHaveAttribute('disabled');
	});

	it('Select matches snapshot', () => {
		const tree = renderer.create(<SelectMock />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Select matches snapshot when tocuhed', () => {
		const tree = renderer
			.create(<SelectMock value="" isInvalid disabled />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
