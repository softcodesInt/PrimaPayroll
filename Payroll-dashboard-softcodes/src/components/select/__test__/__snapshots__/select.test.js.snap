// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component Select matches snapshot 1`] = `
<div
  className="sc-bdnylx gbqhDp"
  data-testid="select"
>
  <label
    htmlFor="select"
  >
    select
  </label>
  <select
    className="form-control"
    id="select"
    onChange={[Function]}
    value="select"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
  <svg>
    arrow-down.svg
  </svg>
</div>
`;

exports[`Select Component Select matches snapshot when tocuhed 1`] = `
<div
  className="sc-bdnylx iPTAEz"
  data-testid="select"
  disabled={true}
>
  <label
    htmlFor="select"
  >
    select
  </label>
  <select
    className="form-control"
    disabled={true}
    id="select"
    onChange={[Function]}
    value=""
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
  <svg>
    arrow-down.svg
  </svg>
  <span
    className="sc-gtssRu kbMXaR error"
    color="#f66262"
    fontSize="12px"
  >
    This field is required
  </span>
</div>
`;
